pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -339,6 +364,15 @@ func (d *DeploymentController) apply(controller string, yml string) error {
 	return nil
 }
 
+func (d *DeploymentController) HandleTagChange(newTags []string) {
+	d.myTags = sets.New(newTags...)
```

Comment:
This is racy w/o mutex.

Personally I would just pass the entire TagWatcher into DeploymentController, have it expose a `watcher.GetTags() sets.String`, and use that everywhere; should simplify things a little bit.

You still need HandleTagChange to enqueue things, though

Reply:
Then `Run` also doesn't need to take it as a param